### PR TITLE
ft: ZENKO-143 transient source

### PR DESCRIPTION
**Pull request by jonathan-gramain**
* cleanup: ZENKO-143 code cleanups
    
    * in ReplicateObject._handleReplicationOutcome(), don't use temporary
      variable 'status'
    
    * make _getGlobalReplicationStatus() a member of ObjectQueueEntry
      class
    
    * cleanups in ObjectQueueEntry.toXxxEntry() helpers

  * ft: ZENKO-143 force read on local source
    
    Make use of x-amz-location-constraint header to force a read from the
    local source, in case a cloud target is the preferred read location
    e.g. w/ transient source.
    
    - If the cloud target replication is PENDING it avoid failing the read
    
    - If it's in COMPLETED status it enforces a local read for transient
    source to be effective.
    
    For this to work, we have to go through a custom backbeat route that
    maps to AWS SDK getObject() with support for non-standard
    x-amz-location-constraint header.

 *  ft: ZENKO-143 garbage-collect transient data locations
    
    After CRR status becomes COMPLETED for an object on a transient source
    location, publish a message to garbage-collect the original data
    locations of the object.

### CODE REVIEW

File: extensions/replication/replicationStatusProcessor/ReplicationStatusProcessor.js [hunk 1]
```diff
@@ -9,6 +9,8 @@ const errors = require('arsenal').errors;
 const BackbeatConsumer = require('../../../lib/BackbeatConsumer');
 const FailedCRRProducer =
     require('../../../extensions/replication/failedCRR/FailedCRRProducer');
```
Comment (philipyoo) [inline extensions/replication/replicationStatusProcessor/ReplicationStatusProcessor.js]: Maybe the file was moved in the past? The path here can be shortened

File: extensions/replication/tasks/UpdateReplicationStatus.js [hunk 2]
```diff
@@ -249,6 +251,15 @@ class UpdateReplicationStatus extends BackbeatTask {
                     replicationStatus:
                         updatedSourceEntry.getReplicationStatus(),
                 });
+                if (config.getIsTransientLocation(
+                    updatedSourceEntry.getDataStoreName()) &&
+                    updatedSourceEntry.getReplicationStatus()
+                    === 'COMPLETED') {
```
Comment (philipyoo) [inline extensions/replication/tasks/UpdateReplicationStatus.js:257]: shouldn't `getIsTransientLocation` take a location name arg?

Comment (jonathan-gramain) [reply extensions/replication/tasks/UpdateReplicationStatus.js:257]: I think it does, confused by the weird indentation? :rofl:

File: extensions/replication/replicationStatusProcessor/ReplicationStatusProcessor.js (hunk 1 above)
Comment (jonathan-gramain) [reply extensions/replication/replicationStatusProcessor/ReplicationStatusProcessor.js]: Yeah, I can silent fix this.

File: extensions/replication/tasks/UpdateReplicationStatus.js (hunk 2 above)
Comment (philipyoo) [reply extensions/replication/tasks/UpdateReplicationStatus.js:257]: Ah my bad, I need glasses lol

**Review by philipyoo — DISMISSED** [review]
Nice!

**Review by philipyoo — DISMISSED** [review]

**Review by jeremyds — DISMISSED** [review]

**Comment by jonathan-gramain** [comment]
@bert-e reset

**Comment by tcarmet** [comment]
@bert-e wait

**Review by philipyoo — APPROVED** [review]

**Review by jeremyds — APPROVED** [review]

**Pull request merged**